pkg/controller/master-controller-manager/external-cluster/cluster_controller.go

```diff
@@ -75,13 +77,34 @@ func Add(
 	}
 
 	// Watch for changes to ExternalCluster
-	return c.Watch(&source.Kind{Type: &kubermaticv1.ExternalCluster{}}, &handler.EnqueueRequestForObject{})
+	return c.Watch(&source.Kind{Type: &kubermaticv1.ExternalCluster{}}, &handler.EnqueueRequestForObject{}, withEventFilter())
+}
+
+func withEventFilter() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			externalCluster, ok := e.Object.(*kubermaticv1.ExternalCluster)
+			if !ok {
+				return false
+			}
+			return externalCluster.Spec.CloudSpec.KubeOne == nil
```

Comment:
Why do we ignore KubeOne clusters during their creation, but reconcile them when they are updated or deleted?